src/main/java/frc/robot/poseestimator/helpers/objectListOrganizer.java

```diff
@@ -0,0 +1,38 @@
+package frc.robot.poseestimator.helpers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class objectListOrganizer<T> {
+
+	private final Supplier<List<T>> observationSupplier;
+	private final int maxDataIntakeRepeats;
+	private List<T> stackedObservations;
+	private int dataIntakeCounter;
+
+	public objectListOrganizer(Supplier<List<T>> observationSupplier, int maxCount) {
+		this.observationSupplier = observationSupplier;
+		this.maxDataIntakeRepeats = maxCount;
+		stackedObservations = new ArrayList<>();
+		dataIntakeCounter = 0;
+	}
+
+	public List<T> getDataIntakelist() {
```

Comment:
I dont like that name

Reply:
you are absulutly right. i just read it out laud